src/AaveV3StrategicAssets_20230606/TestStrategicAssetsManager.t.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {Test} from 'forge-std/Test.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+import {AaveV2Ethereum} from 'aave-address-book/AaveV2Ethereum.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+
+import {StrategicAssetsManager} from './StrategicAssetsManager.sol';
+import {VeTokenManager} from './VeTokenManager.sol';
+import {Core} from './Core.sol';
+
+contract StrategicAssetsManagerTest is Test {
+  event GuardianUpdated(address oldGuardian, address newGuardian);
+  event WithdrawalERC20(address indexed _token, address _to, uint256 _amount);
+
+  // VeToken
+  address public constant B_80BAL_20WETH = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+
+  StrategicAssetsManager public strategicAssets;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 17523941);
+
+    vm.startPrank(AaveGovernanceV2.SHORT_EXECUTOR);
+    strategicAssets = new StrategicAssetsManager();
+    vm.stopPrank();
+  }
+}
+
+contract Initialize is StrategicAssetsManagerTest {
+  function test_revertsIf_alreadyInitialized() public {
+    vm.expectRevert('Initializable: contract is already initialized');
+    strategicAssets.initialize();
+  }
+}
+
+contract TransferOwnership is StrategicAssetsManagerTest {
+  function test_revertsIf_invalidCaller() public {
+    vm.expectRevert('Ownable: caller is not the owner');
+    strategicAssets.transferOwnership(makeAddr('new-admin'));
+  }
+
+  function test_successful() public {
+    address newAdmin = makeAddr('new-admin');
+    vm.startPrank(AaveGovernanceV2.SHORT_EXECUTOR);
+    strategicAssets.transferOwnership(newAdmin);
+    vm.stopPrank();
+
+    assertEq(newAdmin, strategicAssets.owner());
+  }
+}
+
+contract SetStrategicAssetManager is StrategicAssetsManagerTest {
+  function test_revertsIf_invalidCaller() public {
+    vm.expectRevert('ONLY_BY_OWNER_OR_GUARDIAN');
+    strategicAssets.updateGuardian(makeAddr('new-admin'));
+  }
+
+  function test_successful() public {
+    address newManager = makeAddr('new-admin');
+    vm.expectEmit();
+    emit GuardianUpdated(strategicAssets.guardian(), newManager);
+    vm.startPrank(AaveGovernanceV2.SHORT_EXECUTOR);
+    strategicAssets.updateGuardian(newManager);
+    vm.stopPrank();
+
+    assertEq(newManager, strategicAssets.guardian());
+  }
+}
+
+contract RemoveStrategicAssetManager is StrategicAssetsManagerTest {
+  function test_revertsIf_invalidCaller() public {
+    vm.expectRevert('ONLY_BY_OWNER_OR_GUARDIAN');
```

Comment:
```suggestion
    vm.expectRevert('ONLY_BY_GUARDIAN');
```

Reply:
library was updated so owner can also update the guardian.
for example, the DAO no longer wants me as a guardian, the DAO as the owner (short_executor) can remove my address.

Reply:
Good stuff, resolving.